health/memoryRecord.go

```diff
@@ -0,0 +1,69 @@
+package health
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"runtime"
+	"runtime/pprof"
+	"strings"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+)
+
+var _ record = (*memoryRecord)(nil)
+
+type memoryRecord struct {
+	stats    runtime.MemStats
+	filename string
+}
+
+func newMemoryRecord(stats runtime.MemStats, parentFolder string) *memoryRecord {
+	filename := formatMemoryRecordFilename(stats)
+	filename = path.Join(parentFolder, filename)
+
+	return &memoryRecord{
+		stats:    stats,
+		filename: filename,
+	}
+}
+
+func formatMemoryRecordFilename(stats runtime.MemStats) string {
+	timestamp := time.Now().Format("20060102150405")
+	inUse := core.ConvertBytes(stats.HeapInuse)
+	inUse = strings.ReplaceAll(inUse, " ", "_")
+	inUse = strings.ReplaceAll(inUse, ".", "_")
+	filename := fmt.Sprintf("mem__%s__%s.pprof", timestamp, inUse)
+	return filename
+}
+
+func (record *memoryRecord) getFilename() string {
+	return record.filename
+}
+
+func (record *memoryRecord) save() error {
+	filename := path.Join(record.filename)
```

Comment:
so, for each save call we open, write an close the file. This is the safest way to persist the data when the process closing is imminent but is slow. Another way to do this is to open once, keep the pointer as long as you need it and call the runtime.SetFinalizer function to close the file if the process exits with OOM. Save will just write in the file. (this is the pattern used in the logger subsytem)

Reply:
Nice idea. 

In the case of `memoryUsageRecord` we open the file, write the heap profile, close it, then never open it again. And we write many such files (one every second if when memory is critical). Does the recommendation still apply in this case?